### PR TITLE
💥 Role / Directive Refactor

### PR DESCRIPTION
**Pull request by fwkoch**
`markdown-it-myst` is a new package to generically parse roles/directives into tokens. It does a nested parse to tokenize args/options/content as well, passing along both the raw role/directive content and the tokenized content.

This package does not define custom role/directive types; those are defined downstream as AST transforms on content made available here.

This package is the first step in addressing #181

---

# Update

This PR has consolidated the work on `markdown-it-myst` with `myst-parser` #193 and `myst-roles/directives` #206 - it addresses all the refactoring described in #181

### CODE REVIEW

File: .changeset/brown-onions-fix.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'myst-cli': patch
+---
+
+Pull tab/grid/card directives into external packages
```
Comment (rowanc1) [inline .changeset/brown-onions-fix.md:5]: Should this be in here? Did this get picked up somehow accidentally?

Comment (fwkoch) [reply .changeset/brown-onions-fix.md:5]: Hmm, yeah not sure how it got picked up here. This changeset was already resolved on main from this commit: https://github.com/executablebooks/mystjs/commit/037c8d5808077a1f34c4e37b8c4fecdf6ba59031

I'll delete the file from here...

**Comment by rowanc1** [comment]
I think that the only bugs that are left are around the card headers.

image

It would be great to have tests on the remaining directives, but also want to get this in and out the door! I think there is another pass on validation we need to do, but that can come in next, I don't think we actually loose that much?

**Comment by rowanc1** [comment]
Some known things:
- Alignment of images is a regression.
- And better validation of width/height.
- figure classes did a bit more on pulling off 'center' and making that the align key.

Anything else?

I think those can easily come in a validation pass.

**Pull request merged**